src/main/kotlin/it/pagopa/wallet/services/WalletService.kt

```diff
@@ -206,6 +199,28 @@ class WalletService(
             }
     }
 
+    private fun buildResponseSessionData(
+        hostedOrderResponse: Fields,
+        isAPM: Boolean
+    ): SessionWalletCreateResponseSessionDataDto =
+        if (isAPM) {
```

Comment:
I would add a `hostedOrderResponse` NPG response state check here.
For cards we expect response state to be `GDI_VERIFICATION` and for APM `REDIRECTED_TO_EXTERNAL_DOMAIN`, right?

Reply:
Added in latest push :)